src/main/java/goblintserver/GoblintClient.java

```diff
@@ -55,15 +60,25 @@ public boolean connectGoblitClient() {
      * Method for sending the requests to Goblint server.
      */
 
-    public void writeMessageToSocket(String message) throws IOException {
-        ByteBuffer buffer = ByteBuffer.allocate(1024);
-        buffer.clear();
-        buffer.put(message.getBytes());
-        buffer.flip();
-        while (buffer.hasRemaining()) {
-            channel.write(buffer);
-        }
-        log.debug("Message written to socket.");
+    public void writeRequestToSocket(String request) throws IOException {
+        OutputStream stream = Channels.newOutputStream(channel);
```

Comment:
The `OutputStream` should probably be a field, instead of recreating it every time.